### ISSUE TITLE
Docker compose container permission errors

### ISSUE THREAD

**Issue by LukasRdm** [mediagis/nominatim-docker#571]
Docker container made with compose with bind mounts has many permission issues.

I am using docker compose to start the container.
Here is the docker-compose.yml:

version: "3"

x-logging:
  &default-logging
  options:
    max-size: '50m'
    max-file: '10'
  driver: json-file

services:
    nominatim:
        container_name: nominatim
        image: mediagis/nominatim:4.4
        ports:
            - "18080:8080"
        environment:
            PBF_URL: https://download.geofabrik.de/europe-latest.osm.pbf
            REPLICATION_URL: https://download.geofabrik.de/europe-updates/
            NOMINATIM_PASSWORD: secret
            IMPORT_WIKIPEDIA: "true"
            THREADS: 8
            UPDATE_MODE: once
        volumes:
            - type: bind
              source: /data/nominatim/db
              target: /var/lib/postgresql/14/main
            - type: bind
              source: /data/nominatim/flatnode
              target: /nominatim/flatnode
        shm_size: 8gb
        logging:  *default-logging

First error is met at Postgres DB start up. Error says:
`Starting PostgreSQL 14 database server                                                                                                                                                        * Error: The cluster is owned by user id 1026 which does not exist`
This can be overcome by adding this line to compose file:
`        command: bash -c "sudo chown -R postgres:postgres /var/lib/postgresql/14/main && /app/start.sh"`

Second and third error is met when trying to update the Nominatim data. The daily updates didn't come through, so I tried updating the data manually with command:
`docker exec -it nominatim sudo -u nominatim nominatim replication --catch-up`

The second error looks like:
`PermissionError: [Errno 13] Permission denied: '/app/tokenizer`

After manually giving permission with command:
`docker exec -it container_id sudo chown nominatim:nominatim /app`

I try to manually update again.
I get through the second error, but similar error pops up:
`RuntimeError: Open failed for '/nominatim/osmosischange.osc': Permission denied`

I am trying the same command:
`docker exec -it container_id sudo chown nominatim:nominatim /app`
Which should finally finish the update.

**Desktop / Server (please complete the following information):**
 - OS & Version: Ubuntu 24.04
 - Docker Version: 27.0.3
 - Nominatim Version: 4.4.0

**Comment by LukasRdm** [comment]
Yes, after using command chown on folder /app for the second time, the update has successfully finished.

**Comment by LukasRdm** [comment]
The issue might be with using `docker exec -it nominatim sudo -u nominatim nominatim replication --catch-up` without the --project-dir env. When I use `docker exec -it nominatim sudo -u nominatim nominatim replication --catch-up --project-dir /nominatim`

**Comment by LukasRdm** [comment]
I switched to normal volumes which solved the permission issues. Still had trouble with `docker exec -it nominatim sudo -u nominatim nominatim replication --catch-up`, but after using the --project-dir `docker exec -it nominatim sudo -u nominatim nominatim replication --catch-up --project-dir /nominatim` all went well. Automatic updates with parameter UPDATE_MODE=once in docker-compose still doesn't work tho.

**Comment by heikokue** [comment]
I’ve come across the same issues and have identified the following two pitfalls that need to be taken into account when running `docker exec`:
1.  The environment variables specified via `docker run` or in `docker-compose.yml` are only available to the `root` user. However, the nominatim commands must be executed as `nominatim` user. To pass the environment via `docker exec ... sudo -u nominatim`, the sudo option `-E` must be specified. The `-E` is missing in the current [howto.md](howto.md) example.
2. Some `nominatim` commands require you to specify the project directory, unless you are already in that directory when you run the command. It is easy to forget this `--project-dir /nominatim` option. It is therefore advisable to specify this always as the working directory: `docker exec ... -w /nominatim`

Imho, the full start of the command should be:
`docker exec -it -w /nominatim <container> sudo -E -u nominatim nominatim ...`
I just created #659. It also added scripts for easier call nominatim command inside the container.

**Comment by edurenye** [comment]
I'm getting another permission error:
`[ERROR] Control server error: [Errno 13] Permission denied: '/root/.gunicorn'`

**Comment by edurenye** [comment]
My error was fixed by adding:
```
environment:
      HOME: /home/nominatim

```
